### PR TITLE
fixes #9927: pascal_voc_detection_metrics low scores for first catategory in label_map

### LINKED ISSUE #9927: pascal_voc_detection_metrics doesn't work for the first category in label_map
I am following this [tutorial](https://tensorflow-object-detection-api-tutorial.readthedocs.io/en/latest/index.html) to train a Faster R-CNN model for object detection on my own data.

When evaluating using  `python model_main_tf2.py --model_dir=models/MY_MODEL --pipeline_config_path=models/MY_MODEL/pipeline.config --checkpoint_dir=models/MY_MODEL` and Pascal VOC evaluator the first category in my label map is never evaluated correctly.

I have trained the modell for a different number of classes, every time the first category has very low mAP.
The code I'm posting here refers to a model trained for object detection for two classes. The record files have been created with the script from the tutorial.

This is my config file:
`# Faster R-CNN with Resnet-50 (v1)
# Trained on COCO, initialized from Imagenet classification checkpoint

# This config is TPU compatible.

model {
  faster_rcnn {
    num_classes: 2
    image_resizer {
      fixed_shape_resizer {
        width: 1024
        height: 1024
      }
    }
    feature_extractor {
      type: 'faster_rcnn_resnet101_keras'
      batch_norm_trainable: true
    }
    first_stage_anchor_generator {
      grid_anchor_generator {
        scales: [0.25, 0.5, 1.0, 2.0]
        aspect_ratios: [0.5, 1.0, 2.0]
        height_stride: 16
        width_stride: 16
      }
    }
    first_stage_box_predictor_conv_hyperparams {
      op: CONV
      regularizer {
        l2_regularizer {
          weight: 0.0
        }
      }
      initializer {
        truncated_normal_initializer {
          stddev: 0.01
        }
      }
    }
    first_stage_nms_score_threshold: 0.0
    first_stage_nms_iou_threshold: 0.7
    first_stage_max_proposals: 300
    first_stage_localization_loss_weight: 2.0
    first_stage_objectness_loss_weight: 1.0
    initial_crop_size: 14
    maxpool_kernel_size: 2
    maxpool_stride: 2
    second_stage_box_predictor {
      mask_rcnn_box_predictor {
        use_dropout: false
        dropout_keep_probability: 1.0
        fc_hyperparams {
          op: FC
          regularizer {
            l2_regularizer {
              weight: 0.0
            }
          }
          initializer {
            variance_scaling_initializer {
              factor: 1.0
              uniform: true
              mode: FAN_AVG
            }
          }
        }
        share_box_across_classes: true
      }
    }
    second_stage_post_processing {
      batch_non_max_suppression {
        score_threshold: 0.0
        iou_threshold: 0.6
        max_detections_per_class: 100
        max_total_detections: 300
      }
      score_converter: SOFTMAX
    }
    second_stage_localization_loss_weight: 2.0
    second_stage_classification_loss_weight: 1.0
    use_static_shapes: true
    use_matmul_crop_and_resize: true
    clip_anchors_to_image: true
    use_static_balanced_label_sampler: true
    use_matmul_gather_in_matcher: true
  }
}

train_config {
  batch_size: 4
  num_steps: 6000
  sync_replicas: true
  optimizer {
    momentum_optimizer {
      learning_rate {
        cosine_decay_learning_rate {
          learning_rate_base: 0.002
          total_steps: 6000
          warmup_learning_rate: 0.0002
          warmup_steps: 300
          hold_base_rate_steps: 0
        }
      }
      momentum_optimizer_value: 0.9
    }
    use_moving_average: false
  }
  fine_tune_checkpoint: "pre-trained-models/faster_rcnn_resnet101_v1_1024x1024_coco17_tpu-8/checkpoint/ckpt-0"
  startup_delay_steps: 0.0
  replicas_to_aggregate: 8
  max_number_of_boxes: 100
  unpad_groundtruth_tensors: false
  fine_tune_checkpoint_type: "detection"
  use_bfloat16: false
  fine_tune_checkpoint_version: V2
  
# Data augmentation
#  data_augmentation_options {
#    random_horizontal_flip {
#    }
#  }
#  data_augmentation_options {
#    random_rgb_to_gray {
#    }
#  }
#  data_augmentation_options {
#    random_patch_gaussian  {
#    }
#  }
#  data_augmentation_options {
#    random_adjust_brightness {
#    }
#  }
#  data_augmentation_options {
#    random_adjust_contrast {
#    }
#  }
# End of data augmentation
  
}
train_input_reader: {
  label_map_path: "annotations/label_map.pbtxt"
  tf_record_input_reader {
    input_path: "annotations/train.record"
  }
}

eval_config: {
  metrics_set: "pascal_voc_detection_metrics"
  use_moving_averages: false
}

eval_input_reader: {
  label_map_path: "annotations/label_map.pbtxt"
  shuffle: false
  num_epochs: 1
  tf_record_input_reader {
    input_path: "annotations/test.record"
  }
}
`

My label_map:
`item {
    id: 1
    name: 'Signal'
}
item {
    id: 2
    name: 'Schild' # this means sign in german
}`

The results from evaluation are
`I0419 13:21:50.826318 140148733331264 object_detection_evaluation.py:1335] average_precision: 0.004911
I0419 13:21:50.853986 140148733331264 object_detection_evaluation.py:1335] average_precision: 0.802144
INFO:tensorflow:Eval metrics at step 6000
I0419 13:21:50.858681 140148733331264 model_lib_v2.py:988] Eval metrics at step 6000
INFO:tensorflow:        + PascalBoxes_Precision/mAP@0.5IOU: 0.403527
I0419 13:21:50.907640 140148733331264 model_lib_v2.py:991]      + PascalBoxes_Precision/mAP@0.5IOU: 0.403527
INFO:tensorflow:        + PascalBoxes_PerformanceByCategory/AP@0.5IOU/Signal: 0.004911
I0419 13:21:50.909214 140148733331264 model_lib_v2.py:991]      + PascalBoxes_PerformanceByCategory/AP@0.5IOU/Signal: 0.004911
INFO:tensorflow:        + PascalBoxes_PerformanceByCategory/AP@0.5IOU/Schild: 0.802144
I0419 13:21:50.910397 140148733331264 model_lib_v2.py:991]      + PascalBoxes_PerformanceByCategory/AP@0.5IOU/Schild: 0.802144
INFO:tensorflow:        + Loss/RPNLoss/localization_loss: 0.009531
I0419 13:21:50.911445 140148733331264 model_lib_v2.py:991]      + Loss/RPNLoss/localization_loss: 0.009531
INFO:tensorflow:        + Loss/RPNLoss/objectness_loss: 0.003197
I0419 13:21:50.912504 140148733331264 model_lib_v2.py:991]      + Loss/RPNLoss/objectness_loss: 0.003197
INFO:tensorflow:        + Loss/BoxClassifierLoss/localization_loss: 0.022416
I0419 13:21:50.913525 140148733331264 model_lib_v2.py:991]      + Loss/BoxClassifierLoss/localization_loss: 0.022416
INFO:tensorflow:        + Loss/BoxClassifierLoss/classification_loss: 0.027070
I0419 13:21:50.914626 140148733331264 model_lib_v2.py:991]      + Loss/BoxClassifierLoss/classification_loss: 0.027070
INFO:tensorflow:        + Loss/regularization_loss: 0.000000
I0419 13:21:50.915687 140148733331264 model_lib_v2.py:991]      + Loss/regularization_loss: 0.000000
INFO:tensorflow:        + Loss/total_loss: 0.062212
I0419 13:21:50.916763 140148733331264 model_lib_v2.py:991]      + Loss/total_loss: 0.062212`

But both objects can be found by model when using inference on test images!

Can somebody help me how to solve this issue? 
Does it have to do with category ID 1 interfering with background category (ID 0)?

### PR DESCRIPTION
**Pull request by GoldschmittGabriel**
Only add 1 if there is at least one element greater than 0 per row
#9927

# Description

> :memo: Please include a summary of the change. 
>  
> * Please also include relevant motivation and context.  
> * List any dependencies that are required for this change.  

## Type of change

For a new feature or function, please create an issue first to discuss it
with us before submitting a pull request.

Note: Please delete options that are not relevant.

- [x ] Bug fix (non-breaking change which fixes an issue)

## Tests

Evaluate results with metric pascal voc evaluator that has unusual low scores on the first class. 
you can stop on the changed line and see if there are row entries in the ground truth variable with only zeros

See also #9927 

**Test Configuration**:

## Checklist

- [x] I have signed the [Contributor License Agreement](https://github.com/tensorflow/models/wiki/Contributor-License-Agreements).
- [x] I have read [guidelines for pull request](https://github.com/tensorflow/models/wiki/Submitting-a-pull-request).
- [x] My code follows the [coding guidelines](https://github.com/tensorflow/models/wiki/Coding-guidelines).
- [x] I have performed a self [code review](https://github.com/tensorflow/models/wiki/Code-review) of my own code.
- [x] I have commented my code, particularly in hard-to-understand areas.
- [x] My changes generate no new warnings.

### CODE REVIEW

**Comment by diaconumadalina** [comment]
Hello everyone, I'm Mada, and I'm eager to begin contributing to open-source projects. TensorFlow seems like an excellent place to start. Could someone please help me determine if a specific issue is still open, and if so, guide me on how to proceed with it? I appreciate any assistance provided.

**Comment by DEV4INO** [comment]
@diaconumadalina sorry for my late response
To be honest, I’m not entirely sure. This issue occasionally appears in relation to #9927, and I first encountered it while working on an AI model about two years ago. If I have the opportunity, I could implement some tests to verify this. However, I’m not certain if this is still a requirement, given that the object detection component seems somewhat outdated. The most recent model added is mobilenetv3small.
